test/e2e/flask/multichain-api/create-session.spec.ts

```diff
@@ -46,4 +61,144 @@ describe('Multichain API', function () {
+
+          await openMultichainDappAndConnectWalletWithExternallyConnectable(
+            driver,
+            extensionId,
+          );
+
+          await addRequestAccountsToCreateSession(driver, [
+            DEFAULT_FIXTURE_ACCOUNT,
+            INVALID_ACCOUNT,
+          ]);
+
+          await initCreateSessionScopes(driver, requestScopes);
+
+          // navigate to network selection screen
+          const editButtons = await driver.findElements('[data-testid="edit"]');
+          await editButtons[1].click();
+          await driver.delay(largeDelayMs);
+
+          await assertOnlyRequestedNetworksAreSelected(
+            driver,
+            networksToRequest,
+          );
+
+          // proceed with network selection confirm
+          await driver.clickElement(
+            '[data-testid="connect-more-chains-button"]',
+          );
+          await confirmAndSwitchFocusToDapp(driver);
+
+          const getSessionScopesResult = await getSessionScopes(driver);
+          for (const scope of requestScopes) {
+            /**
+             * Accounts in scope should not include invalid account {@link INVALID_ACCOUNT}, only the valid accounts.
+             */
+            const expectedSessionScope = getExpectedSessionScope(scope, [
+              DEFAULT_FIXTURE_ACCOUNT,
+            ]);
+            const result = getSessionScopesResult.sessionScopes[scope].accounts;
+
+            assert.deepEqual(
+              expectedSessionScope.accounts,
+              result,
+              `${expectedSessionScope.accounts} does not match accounts in scope ${result}`,
+            );
+          }
+        },
+      );
+    });
+  });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_createSession` with requested EVM scope that match the user’s enabled networks, edit selection in wallet UI', function () {
+    it('should change result according to changed network & accounts', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          fixtures: new FixtureBuilder()
+            .withPopularNetworks()
+            .withPreferencesControllerAdditionalAccountIdentities()
+            .build(),
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          const requestScopesToNetworkMap = {
+            'eip155:1': 'Ethereum Mainnet',
+            'eip155:10': 'OP Mainnet',
+          };
+
+          const requestScopes = Object.keys(requestScopesToNetworkMap);
+
+          await openMultichainDappAndConnectWalletWithExternallyConnectable(
+            driver,
+            extensionId,
+          );
+
+          await addRequestAccountsToCreateSession(driver, [
+            DEFAULT_FIXTURE_ACCOUNT,
+            '',
```

Comment:
is this empty on purpose?

Reply:
yes, this method takes in `[string, string]` for easier explicitness inside it's body, so I had to pass in a second string arg in this array. The test dapp will filter out empty strings though so in the end it won't have any unpredictable behaviour.